numpy/random/mtrand/mtrand.pyx

```diff
@@ -158,7 +158,9 @@ cdef object cont0_array(rk_state *state, rk_cont0 func, object size,
     cdef npy_intp i
 
     if size is None:
-        return func(state)
+        with lock, nogil:
+            rv = func(state)
```

Comment:
I think that's a deliberate choice, not a bug. Either way, it's certainly a good reason to write the code the way you did. LGTM